src/replica/replica_chkpt.cpp

```diff
@@ -191,46 +198,29 @@ void replica::init_checkpoint(bool is_emergency)
         _stub->_counter_recent_trigger_emergency_checkpoint_count->increment();
 }
 
-// @ secondary
-void replica::on_copy_checkpoint(const replica_configuration &request,
-                                 /*out*/ learn_response &response)
+// ThreadPool: THREAD_POOL_REPLICATION
+void replica::on_query_last_checkpoint(/*out*/ learn_response &response)
 {
     _checker.only_one_thread_access();
 
-    if (request.ballot > get_ballot()) {
-        if (!update_local_configuration(request)) {
-            response.err = ERR_INVALID_STATE;
-            return;
-        }
-    }
-
-    if (status() != partition_status::PS_SECONDARY) {
-        response.err = ERR_INVALID_STATE;
-        return;
-    }
-
     if (_app->last_durable_decree() == 0) {
-        response.err = ERR_OBJECT_NOT_FOUND;
+        response.err = ERR_PATH_NOT_FOUND;
         return;
     }
 
     blob placeholder;
     int err = _app->get_checkpoint(0, placeholder, response.state);
```

Comment:
I suggest you can refactor function `get_checkpoint` in your further pull request, it seems only the last parameter useful, you can delete the first two parameters of this function.

Reply:
get_checkpoint is `pegasus` implementation

Reply:
Yes, I have read the code of `get_checkpoint`, the first two parameters of this function is useless, you can refactor it in future.